netmap/policy.go

```diff
@@ -30,6 +30,46 @@ type PlacementPolicy struct {
 	replicas []netmap.Replica
 }
 
+func copyFilter(f netmap.Filter) netmap.Filter {
+	var filter netmap.Filter
+
+	filter.SetName(f.GetName())
+	filter.SetKey(f.GetKey())
+	filter.SetOp(f.GetOp())
+	filter.SetValue(f.GetValue())
+
+	filters := make([]netmap.Filter, len(f.GetFilters()))
+	for i, internalFilter := range f.GetFilters() {
+		filters[i] = copyFilter(internalFilter)
+	}
+
+	filter.SetFilters(filters)
+
+	return filter
+}
+
+// CopyTo writes deep copy of the [PlacementPolicy] to dst.
+func (p PlacementPolicy) CopyTo(dst *PlacementPolicy) {
+	if dst == nil {
+		return
+	}
+
+	dst.SetContainerBackupFactor(p.backupFactor)
+
+	dst.filters = make([]netmap.Filter, len(p.filters))
+	for i, f := range p.filters {
+		dst.filters[i] = copyFilter(f)
+	}
+
+	// netmap.Selector is a struct with simple types, no links inside. Just create a new slice and copy all items inside.
+	dst.selectors = make([]netmap.Selector, 0, len(p.selectors))
```

Comment:
you know length in advance, so i'd expect `copy` (while this one is also correct)